src/callback-invocation.h

```diff
@@ -44,8 +44,17 @@ struct callback_invocation_holder
 
     ~callback_invocation_holder()
     {
-        if( invocation )
-            owner->deallocate( invocation );
+        if (invocation)
```

Comment:
Should also check `owner`.
Please use agreed formatting - no space after `if`, spaces in parentheses

Reply:
Checking `owner` for now.